src/lib.rs

```diff
@@ -359,42 +359,46 @@ macro_rules! py_run_impl {
 
 /// Test readme and user guide
 #[doc(hidden)]
+#[cfg(feature = "macros")]
 pub mod doc_test {
     macro_rules! doc_comment {
-        ($x:expr, $module:item) => {
-            #[doc = $x]
+        ($doc:expr, $module:item) => {
+            #[doc = $doc]
+            #[allow(non_snake_case)]
             $module
         };
     }
 
     macro_rules! doctest {
-        ($x:expr, $y:ident) => {
-            doc_comment!(include_str!($x), mod $y {});
+        ($($path:ident)/+.md) => {
+            $crate::paste::paste!(
+                doc_comment!(
+                    include_str!(concat!("..", $("/", stringify!($path)),+, ".md")),
+                    mod [<$($path)_+ _md>] {}
+                );
+            );
```

Comment:
Why do we need this?
I prefer to keep this simple for future contributors.

Reply:
Also, why is `paste!` necessary here?

Reply:
I was using some trickery with `paste!` to generate the module names from the path - paste is used to join the idents into one.

I thought it would be nice to simplify usage of the macros. TBH I was 50/50 if this was a nice result in the end, so I'll revert this change.